pkg/docker/config/config.go

```diff
@@ -0,0 +1,271 @@
+package config
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/containers/image/types"
+	helperclient "github.com/docker/docker-credential-helpers/client"
+	"github.com/docker/docker/pkg/homedir"
+	"github.com/pkg/errors"
+)
+
+type dockerAuthConfig struct {
+	Auth string `json:"auth,omitempty"`
+}
+
+type dockerConfigFile struct {
+	AuthConfigs map[string]dockerAuthConfig `json:"auths"`
+	CredHelpers map[string]string           `json:"credHelpers,omitempty"`
+}
+
+const (
+	authCfg           = ".config/containers"
+	authCfgFileName   = "auth.json"
+	dockerCfg         = ".docker"
+	dockerCfgFileName = "config.json"
+)
+
+var (
+	// ErrNotLoggedIn is returned for users not logged into a registry
+	// that they are trying to logout of
+	ErrNotLoggedIn = errors.Errorf("not logged in")
+)
+
+// SetAuthentication stores the base64 encoding of username and password
+// in the auth file if the credentials are authenticated
+func SetAuthentication(username, password, registry string) error {
```

Comment:
(This would feel a bit more natural to me with parameters ordered (registry, username, password), similar to `registry[username] = password`, and from more general to more specific. An absolutely non-blocking matter of taste.)

Reply:
(Is ”authentication” the best way to describe what is being stored? Usually ”authentication” means the _process_ of verifying the client is who it claims to be. ”Credentials” perhaps?

OTOH I am not a native speaker, and the config format does use `auths`. Not sure.  Maybe the public API should say credentials / username+password, which the caller cares about, and the private helpers, variables and data types should continue to say `auth` to match the config file.)

Reply:
fixed